### PR TITLE
RC config poller with multiple listeners per product

### PR DESCRIPTION
**Pull request by ValentinZakharov**
# What Does This Do
Improved Configuration poller to support multiple listeners per product.

# Motivation
The changes needed to follow implementation of Api Security Sample Rate configuration via remote config.

# Additional Notes
This is preparation for https://github.com/DataDog/dd-trace-java/pull/6340

### CODE REVIEW

File: remote-config/src/main/java/datadog/remoteconfig/tuf/RemoteConfigRequest.java [hunk 1]
```diff
@@ -62,6 +62,7 @@ public static class ClientInfo {
     public static final long CAPABILITY_ASM_CUSTOM_RULES = 1 << 8;
     public static final long CAPABILITY_ASM_CUSTOM_BLOCKING_RESPONSE = 1 << 9;
     public static final long CAPABILITY_ASM_TRUSTED_IPS = 1 << 10;
+    public static final long CAPABILITY_ASM_API_SECURITY_SAMPLE_RATE = 1 << 11;
```
Comment (smola) [inline remote-config/src/main/java/datadog/remoteconfig/tuf/RemoteConfigRequest.java:65]: This does not belong to this PR.

File: remote-config/src/main/java/datadog/remoteconfig/ConfigurationPoller.java [hunk 2]
```diff
@@ -129,7 +129,9 @@ public ConfigurationPoller(
   }
 
   public synchronized <T> void addListener(Product product, ProductListener listener) {
```
Comment (smola) [inline remote-config/src/main/java/datadog/remoteconfig/ConfigurationPoller.java:131]: Do we use `removeListener` in prod anywhere? Given that `removeListener` is now `remoteListeners` (it removes all listeners), is there any unexpected inteteraction from this?

File: remote-config/src/main/java/datadog/remoteconfig/tuf/RemoteConfigRequest.java (hunk 1 above)
Comment (ValentinZakharov) [reply remote-config/src/main/java/datadog/remoteconfig/tuf/RemoteConfigRequest.java]: Thanks, that's from another PR. Removed

File: remote-config/src/main/java/datadog/remoteconfig/ConfigurationPoller.java (hunk 2 above)
Comment (ValentinZakharov) [reply remote-config/src/main/java/datadog/remoteconfig/ConfigurationPoller.java]: Renamed method

**Review by smola — APPROVED** [review]
